### PR TITLE
React frontend improvements

### PR DESCRIPTION
**Pull request by Delodax**
- Clickable headers
- Minor restructuring of table markup
- Coloring of selected columns
- Removed React "missing keys" warnings

### CODE REVIEW

File: client/TableData.js [hunk 1]
```diff
@@ -30,10 +30,10 @@ const TableData = ({ modules, courseId }) => {
   }
 
   return (<tbody>
-    {submissions.map((s) =>
-      (<tr>
-        <td>{s.user.name}</td>
-        {s.submissions.map((cell) => (<td className='grade-column'>{cell.grade}</td>))}
+    {submissions.map((s, sIndex) =>
+      (<tr key={sIndex}>
```
Comment (exacs) [inline client/TableData.js]: I think it's better if we replace this `sIndex` (the "position of array") to an actual "key" that identifies the row, for example some user ID

File: client/TableData.js [hunk 2]
```diff
@@ -30,10 +30,10 @@ const TableData = ({ modules, courseId }) => {
...
+        <th className='first-header-column' scope='row'>{s.user.name}</th>
+        {s.submissions.map((cell, cellIndex) => (<td key={`${sIndex}-${cellIndex}`} className='grade-column'>{cell.grade}</td>))}
```
Comment (exacs) [inline client/TableData.js]: Same here. If we can find a "key" that identifies the cell (a user-assignment combo or submission or something) would be nice!

File: client/TableData.js (hunk 1 above)
Comment (Delodax) [reply client/TableData.js]: I agree. I resorted to using the (subpar) indexes at a couple of places where no obvious alternatives were present. I'll look into another solution!

File: client/TableData.js (hunk 2 above)
Comment (Delodax) [reply client/TableData.js]: Agreed

**Pull request merged**